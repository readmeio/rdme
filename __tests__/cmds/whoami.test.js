const config = require('config');
const configStore = require('../../src/lib/configstore');
const Command = require('../../src/cmds/whoami');

const cmd = new Command();

describe('rdme whoami', () => {
  it('should error if user is not authenticated', () => {
    configStore.delete('email');
    configStore.delete('project');

    return expect(cmd.run({})).rejects.toStrictEqual(new Error(`Please login using \`${config.get('cli')} login\`.`));
  });

  it('should return the authenticated user', () => {
    configStore.set('email', 'email@example.com');
    configStore.set('project', 'subdomain');

    return expect(cmd.run({})).resolves.toMatchSnapshot();
  });
});
