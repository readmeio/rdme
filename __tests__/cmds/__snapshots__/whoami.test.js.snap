// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rdme whoami should return the authenticated user 1`] = `"You are currently logged in as email@example.com to the subdomain project."`;
